src/main/java/org/opentripplanner/street/model/edge/EscalatorEdge.java

```diff
@@ -0,0 +1,32 @@
+package org.opentripplanner.street.model.edge;
+
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.street.model.vertex.Vertex;
+import org.opentripplanner.street.search.TraverseMode;
+import org.opentripplanner.street.search.state.State;
+
+/** Represents an escalator. An escalator edge can only be traversed by walking */
+public class EscalatorEdge extends Edge {
+
+  public EscalatorEdge(Vertex v1, Vertex v2) {
+    super(v1, v2);
+  }
+
+  @Override
+  public State[] traverse(State s0) {
+    // Only allow traversal by walking
+    if (s0.getNonTransitMode() == TraverseMode.WALK) {
+      var s1 = s0.edit(this);
+      var time = getDistanceMeters() / s0.getPreferences().walk().speed();
```

Comment:
I wonder if we should consider that walk speed preference here at all? We could just have some static (maybe configurable) speed. I wonder if the speed is ever set through tagging in OSM.

Reply:
You could try to search the internet if you can find what is usually the horizontal velocity of escalators and just use that as a static speed for the escalator instead of using the walk speed from preferences.